In [9]:
import pandas as pd
import numpy as np
df=pd.read_csv("Students Social Media Addiction.csv")

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 705 entries, 0 to 704
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Student_ID                    705 non-null    int64  
 1   Age                           705 non-null    int64  
 2   Gender                        705 non-null    object 
 3   Academic_Level                705 non-null    object 
 4   Country                       705 non-null    object 
 5   Avg_Daily_Usage_Hours         705 non-null    float64
 6   Most_Used_Platform            705 non-null    object 
 7   Affects_Academic_Performance  705 non-null    object 
 8   Sleep_Hours_Per_Night         705 non-null    float64
 9   Mental_Health_Score           705 non-null    int64  
 10  Relationship_Status           705 non-null    object 
 11  Conflicts_Over_Social_Media   705 non-null    int64  
 12  Addicted_Score                705 non-null    int64  
dtypes: fl

In [11]:
print(df.isnull().sum())

Student_ID                      0
Age                             0
Gender                          0
Academic_Level                  0
Country                         0
Avg_Daily_Usage_Hours           0
Most_Used_Platform              0
Affects_Academic_Performance    0
Sleep_Hours_Per_Night           0
Mental_Health_Score             0
Relationship_Status             0
Conflicts_Over_Social_Media     0
Addicted_Score                  0
dtype: int64


In [21]:
# 1. Overall Summary for Numerical Data
print("--- 1. Statistical Summary of Numerical Features ---")
print(df.describe())

# 2. Count of Categorical Data
print("\n--- 2. Count of Students by Academic_Level ---")
print(df['Academic_Level'].value_counts())

print("\n--- 3. Count of Students by Most_Used_Platform ---")
print(df['Most_Used_Platform'].value_counts())
      
print("\n--- 4. Count of Students by Gender ---")
print(df['Gender'].value_counts())  

--- 1. Statistical Summary of Numerical Features ---
       Student_ID         Age  Avg_Daily_Usage_Hours  Sleep_Hours_Per_Night  \
count  705.000000  705.000000             705.000000             705.000000   
mean   353.000000   20.659574               4.918723               6.868936   
std    203.660256    1.399217               1.257395               1.126848   
min      1.000000   18.000000               1.500000               3.800000   
25%    177.000000   19.000000               4.100000               6.000000   
50%    353.000000   21.000000               4.800000               6.900000   
75%    529.000000   22.000000               5.800000               7.700000   
max    705.000000   24.000000               8.500000               9.600000   

       Mental_Health_Score  Conflicts_Over_Social_Media  Addicted_Score  \
count           705.000000                   705.000000      705.000000   
mean              6.226950                     2.849645        6.436879   
std       

In [7]:
# Analyze the average score and sleep grouped by Gender
gender_summary = df.groupby('Gender')[['Addicted_Score', 'Sleep_Hours_Per_Night']].mean()

print("\n--- Average Score and Sleep Hours by Gender ---")
print(gender_summary)

# Analyze the usage distribution by Academic Level
usage_by_academic = df.groupby('Academic_Level')['Avg_Daily_Usage_Hours'].agg(['mean', 'median', 'std'])

print("\n--- Usage Hour Metrics Grouped by Academic_Level ---")
print(usage_by_academic.sort_values(by='mean', ascending=False))


--- Average Score and Sleep Hours by Gender ---
        Addicted_Score  Sleep_Hours_Per_Night
Gender                                       
Female        6.515581               6.816147
Male          6.357955               6.921875

--- Usage Hour Metrics Grouped by Academic_Level ---
                    mean  median       std
Academic_Level                            
High School     5.544444     5.6  0.690225
Undergraduate   5.001416     4.7  1.297057
Graduate        4.776923     4.7  1.228465


In [15]:

# Group by 'Relationship Status' and calculate the mean, median, and count 
# of the 'Addicted_Score' for each status.
relationship_analysis = df.groupby('Relationship_Status')['Addicted_Score'].agg(
    ['mean', 'median', 'count']
)

relationship_analysis_sorted = relationship_analysis.sort_values(by='mean', ascending=False)

print("--- Addiction Score Metrics Grouped by Relationship Status ---")
print(relationship_analysis_sorted)

--- Addiction Score Metrics Grouped by Relationship Status ---
                         mean  median  count
Relationship_Status                         
Complicated          7.031250     7.0     32
Single               6.458333     7.0    384
In Relationship      6.342561     6.0    289


In [8]:
# 1. Pivot Table: Average Addiction Score by Relationship Status and Gender
pivot_table_results = pd.pivot_table(
    data=df,
    values='Addicted_Score',
    index='Relationship_Status',
    columns='Gender',
    aggfunc='mean',
    fill_value='N/A'
)
print("\n--- Average Addiction Score by Relationship_Status and Gender ---")
print(pivot_table_results)

# 2. Correlation Analysis
numerical_cols = ['Age', 'Avg_Daily_Usage_Hours', 'Sleep_Hours_Per_Night', 'Addicted_Score']
correlation_with_target = df[numerical_cols].corr()['Addicted_Score'].sort_values(ascending=False)

print("\n--- Correlation of Numerical Factors with Addiction_Score ---")
print(correlation_with_target)


--- Average Addiction Score by Relationship_Status and Gender ---
Gender                 Female      Male
Relationship_Status                    
Complicated          7.470588  6.533333
In Relationship      6.673913  6.039735
Single               6.323232  6.602151

--- Correlation of Numerical Factors with Addiction_Score ---
Addicted_Score           1.000000
Avg_Daily_Usage_Hours    0.832000
Age                     -0.166396
Sleep_Hours_Per_Night   -0.764858
Name: Addicted_Score, dtype: float64


In [22]:
print(df.columns.tolist())

['Student_ID', 'Age', 'Gender', 'Academic_Level', 'Country', 'Avg_Daily_Usage_Hours', 'Most_Used_Platform', 'Affects_Academic_Performance', 'Sleep_Hours_Per_Night', 'Mental_Health_Score', 'Relationship_Status', 'Conflicts_Over_Social_Media', 'Addicted_Score', 'Gender_Encoded']


In [24]:
import pandas as pd
import numpy as np

df['Avg_Daily_Usage_Hours'] = df['Avg_Daily_Usage_Hours'].replace(['-', '—'], np.nan)
df['Avg_Daily_Usage_Hours '] = pd.to_numeric(df['Avg_Daily_Usage_Hours'], errors='coerce')
df['Avg_Daily_Usage_Hours'].fillna(df['Avg_Daily_Usage_Hours'].mean(), inplace=True)


if 'Student_ID' in df.columns:
    df.drop(columns=['Student_ID'], inplace=True, errors='ignore')

# Encode Binary Feature (Gender)
df['Gender_Encoded'] = df['Gender'].map({'Male': 1, 'Female': 0})

# One-Hot Encode Multi-Category Features (Crucial: Relationship Status)
cols_to_onehot = ['Academic_Level', 'Country', 'Most_Used_Platform', 'Affects_Academic_Performance', 'Relationship_Status']

# THIS CREATES df_processed with the new numeric columns
df_processed = pd.get_dummies(df, columns=cols_to_onehot, drop_first=True)

# Final Cleanup: Drop original text columns
cols_to_drop_final = ['Gender'] 
df_processed.drop(columns=cols_to_drop_final, inplace=True, errors='ignore')

print("Data processing successful. df_processed is ready.")

C:\Users\user\AppData\Local\Temp\ipykernel_10128\3542506161.py:7: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Avg_Daily_Usage_Hours'].fillna(df['Avg_Daily_Usage_Hours'].mean(), inplace=True)


Data processing successful. df_processed is ready.


In [29]:
# 1. Identify all new One-Hot Encoded Relationship Status columns
relationship_cols = [col for col in df_processed.columns if 'Relationship Status' in col]

# 2. Calculate the correlation matrix
cols_to_correlate = relationship_cols + ['Addicted_Score']
correlation_matrix = df_processed[cols_to_correlate].corr()

# 3. Extract and sort the correlation values with Addicted_Score
status_correlation = correlation_matrix['Addicted_Score'].drop('Addicted_Score').sort_values(ascending=False)

print("\n--- Correlation of Individual Relationship Status Categories with Addicted Score ---")
print(status_correlation.to_string())


--- Correlation of Individual Relationship Status Categories with Addicted Score ---
Series([], )


#PROJECT FINDINGS #

1)The data set contain total 705 students among them 353 are undegraduate, 325 are graduate and 27 high school students
2)Addicted score has a mean 6.43,suggesting moderatly high levels of social media dependency
3)Most used social meadia platform is Instagram.
4)When we analyse the academic level and usage hours there is a clear decreasing trend in average usage as the academic level progresses.Hight school>Undergraduate>Graduates>
5)While analysing relationship status and gender by addiction,females with complicated relationship have highest addiction score
6)The factors sleep hours and avg daily usage are mostly influencing the addiction score.
7)From the scatter diagram we can see that the individual who report a higher level of mental health tend to a lower addiction.

#Conclusion#
The analysis strongly suggests that excessive daily usage and its related consequences, specifically poor sleep and lower mental health, are the most significant indicators of a high Addiction Score. Which leades to their academic performance. These risks appear highest among High School students and certain demographic subgroups, particularly females in complicated/committed relationships and single males.
